### PR TITLE
Library support

### PR DESCRIPTION
**Pull request by dapp**
This track of work adds both support for building libraries with blueprint, and new tools for building apps with a combination of components provided by both an app and library layer.

### CODE REVIEW

File: architecture/build.gradle [hunk 1]
```diff
@@ -69,9 +69,11 @@ uploadArchives {
...
             beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
 
-            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                authentication(userName: sonatypeUsername, password: sonatypePassword)
-            }
+//            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+//                authentication(userName: sonatypeUsername, password: sonatypePassword)
+//            }
+
+            repository(url: "file:///Users/mdappo200/.m2/repository/")
```
Comment (fkose696) [inline architecture/build.gradle:76]: can you revert these (applies to all gradle files)

File: sample-library-app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <activity android:name="com.xfinity.blueprint_sample_library_app.LauncherActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN"/>
+
+                <category android:name="android.intent.category.LAUNCHER"/>
+            </intent-filter>
+        </activity>
+        <activity android:name="com.xfinity.blueprint_sample_library_app.StaticScreenActivity" android:label="@string/static_screen"/>
+        <activity android:name="com.xfinity.blueprint_sample_library_app.DynamicScreenActivity" android:label="@string/dynamic_screen"/>
+        <activity android:name="com.xfinity.blueprint_sample_library.ArchitectSampleActivity" android:label="@string/bp_arch_screen"/>
```
Comment (fkose696) [inline sample-library-app/src/main/AndroidManifest.xml:34]: this should be com.xfinity.blueprint_sample_library_app.ArchitectSampleActivity

File: library/build.gradle [hunk 3]
```diff
@@ -52,10 +52,10 @@ task javadoc(type: Javadoc) {
...
 }
 
-task javadocJar(type: Jar, dependsOn: javadoc) {
-    classifier = 'javadoc'
-    from tasks.javadoc.destinationDir
-}
+//task javadocJar(type: Jar, dependsOn: javadoc) {
+//    classifier = 'javadoc'
+//    from tasks.javadoc.destinationDir
+//}
```
Comment (alekhyathandlay) [inline library/build.gradle:58]: cleanup

File: library/build.gradle [hunk 4]
```diff
@@ -81,6 +81,8 @@ uploadArchives {
                 authentication(userName: sonatypeUsername, password: sonatypePassword)
             }
 
+            repository(url: "file:///Users/mdappo200/.m2/repository/")
```
Comment (alekhyathandlay) [inline library/build.gradle:84]: is this change intentional

**Review by alekhyathandlay — COMMENTED** [review]
Looks good. Just wondering about this addition - repository(url: "file:///Users/mdappo200/.m2/repository/")

File: library/build.gradle (hunk 3 above)
Comment (dapp) [reply library/build.gradle]: Fixed

File: library/build.gradle (hunk 4 above)
Comment (dapp) [reply library/build.gradle]: Fixed

File: sample-library-app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (dapp) [reply sample-library-app/src/main/AndroidManifest.xml]: Fixed

File: architecture/build.gradle (hunk 1 above)
Comment (dapp) [reply architecture/build.gradle]: Fixed

File: app/build.gradle [hunk 5]
```diff
@@ -1,7 +1,19 @@
+plugins {
+    id 'com.xfinity.resourceprovider' version "1.4.0"
```
[the remark this answers is not in the document]
Comment (dapp) [reply app/build.gradle]: Fixed

**Review by alekhyathandlay — APPROVED** [review]
Yay

**Review by fkose696 — APPROVED** [review]

**Review by fkose696 — APPROVED** [review]
